File: Treachery-iOS/Treachery-iOS/Models/Rarity.swift
//
//  Rarity.swift
//  Treachery-iOS
//
//  Created by Luke Solomon on 3/12/26.
//

import Foundation

enum Rarity: String, Codable, CaseIterable, Identifiable {
    case uncommon
    case rare
    case mythic
    case special

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .uncommon: return "Uncommon"
        case .rare:     return "Rare"
        case .mythic:   return "Mythic"
        case .special:  return "Special"
        }
    }
}

File: TreacheryAndroid/app/src/main/java/com/solomon/treachery/model/Rarity.kt
package com.solomon.treachery.model

enum class Rarity(val value: String) {
    UNCOMMON("uncommon"),
    RARE("rare"),
    MYTHIC("mythic"),
    SPECIAL("special");

    val displayName: String
        get() = when (this) {
            UNCOMMON -> "Uncommon"
            RARE -> "Rare"
            MYTHIC -> "Mythic"
            SPECIAL -> "Special"
        }

    companion object {
        fun fromValue(value: String): Rarity? = entries.find { it.value == value }
    }
}
